packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/test/Providers/ClientProviders/ClientProviderTests.cs

```diff
@@ -103,17 +103,17 @@ public void TestBuildFields(List<InputParameter> inputParameters, bool containsA
             // validate other optional parameters as fields
             if (containsAdditionalOptionalParams)
             {
-                var optionalParamField = fields.FirstOrDefault(f => f.Name == "_optionalParam");
+                var optionalParamField = fields.FirstOrDefault(f => f.Name == "_optionalNullableParam");
                 Assert.IsNotNull(optionalParamField);
-                Assert.AreEqual(new CSharpType(typeof(string)), optionalParamField?.Type);
+                Assert.AreEqual(new CSharpType(typeof(string), isNullable: true), optionalParamField?.Type);
 
                 var optionalParam2Field = fields.FirstOrDefault(f => f.Name == "_optionalParam2");
                 Assert.IsNotNull(optionalParam2Field);
-                Assert.AreEqual(new CSharpType(typeof(string)), optionalParam2Field?.Type);
+                Assert.AreEqual(new CSharpType(typeof(string), isNullable: false), optionalParam2Field?.Type);
```

Comment:
this looks weird.
An optional parameter, its type should be nullable - is it not? It is quite peculiar that here we have `isNullable` false